x-pack/plugins/infra/public/containers/logs/log_analysis/log_analysis_module_configuration.ts

```diff
@@ -44,7 +50,8 @@ export const isJobConfigurationOutdated = <JobType extends string>(
       new Set(jobConfiguration.indexPattern.split(',')),
       new Set(currentSourceConfiguration.indices)
     ) &&
-    jobConfiguration.timestampField === currentSourceConfiguration.timestampField
+    jobConfiguration.timestampField === currentSourceConfiguration.timestampField &&
+    equal(datafeedRuntimeMappings, currentSourceConfiguration.runtimeMappings)
```

Comment:
💬
This is a little different to the index patterns check, which checks for a subset. Something is only flagged if the current source configuration doesn't contain something from the settings stored with the job.

Whereas here it's a full comparison. I think this makes more sense due to the nature of the mappings and the fact names and scripts can change. But I'm not 100% sure.

Reply:
I would agree. Initially I thought we might be able to limit the comparison to the field names we know are being used. But since runtime fields can depend on each other we don't really know which fields are involved. So a comprehensive check seems safer. It might lead to more noise in the UI, but we could mitigate that by improving the error message in the future (e.g. list the diff of the job configuration in some way).

Reply:
I would agree. Initially I thought we might be able to limit the comparison to the field names we know are being used. But since runtime fields can depend on each other we don't really know which fields are involved. So a comprehensive check seems safer. It might lead to more noise in the UI, but we could mitigate that by improving the error message in the future (e.g. list the diff of the job configuration in some way).

Reply:
> It might lead to more noise in the UI, but we could mitigate that by improving the error message in the future (e.g. list the diff of the job configuration in some way).

Agreed. This sounds like a good / helpful followup.